source/serial_api.c

```diff
@@ -227,9 +235,26 @@ void serial_putc(serial_t *obj, int c) {
     UART_HAL_Putchar(obj->serial.address, (uint8_t)c);
 }
 
+/** This function has been introduced due to a HW bug
+ *  in the Kinetis UART module: after an overrun in the
+ *  FIFO occurs (a character isn't fetched by IRQ when
+ *  it is already full and a new one arrives), the FIFO
+ *  stops working; write and read pointers of the FIFO
+ *  will have an offset afterwards.
+ */
+static void serial_overrun_reset(uint32_t addr)
+{
+    HW_UART_PFIFO_CLR(addr, UART_PFIFO_RXFE_MASK);
+    HW_UART_CFIFO_SET(addr, UART_CFIFO_RXFLUSH_MASK);
```

Comment:
The flush is there because the manual says that flush should be performed after "changing the flag".